src/Model/Contact.php

```diff
@@ -1913,7 +1913,14 @@ public static function updateAvatar(int $cid, string $avatar, bool $force = fals
 			$avatar = self::getDefaultAvatar($contact, Proxy::SIZE_SMALL);
 		}
 
-		if (in_array($contact['network'], [Protocol::FEED, Protocol::MAIL]) || DI::config()->get('system', 'cache_contact_avatar')) {
+		$cache_avatar = DI::config()->get('system', 'cache_contact_avatar');
+
+		// Local contact avatars don't need to be cached
+		if ($cache_avatar && Network::isLocalLink($contact['url'])) {
+			$cache_avatar = !DBA::exists('contact', ['nurl' => $contact['nurl'], 'self' => true]);
+		}
```

Comment:
This can be rewritten as a one-liner:
```suggestion
		// Local contact avatars don't need to be cached
		$cache_avatar = 
			DI::config()->get('system', 'cache_contact_avatar')
                         && Network::isLocalLink($contact['url'])
			&& !DBA::exists('contact', ['nurl' => $contact['nurl'], 'self' => true]);
```

Reply:
I know. But this doesn't really increase readability.

Reply:
I disagree, in my version the `$cache_avatar` flag is only set once, and never accessed before it's actually useful.

Reply:
I'm not so much in multi-line "one-liners" because I always lose the focus when I'm trying to read them. Yes, the variable is only written once in your version, but I think my server is more readable.

Reply:
I guess we need a third pair of eyes then. @nupplaphil @tobiasd @fabrixxm ?

Reply:
For me annandos version is more readable. The compacted version has lost ease of reading by connecting all the boolean statements into one.

Reply:
While at first I would prefer @MrPetovan version, on a secon read @annando is more clear on the intent to me.

Reply:
Thanks, folks!